opentelemetry/src/sdk/trace/span.rs

```diff
@@ -182,6 +191,13 @@ impl Drop for SpanInner {
     }
 }
 
+impl Drop for SpanInner {
+    /// Report span on inner drop
+    fn drop(&mut self) {
+        self.finish_and_process();
```

Comment:
Nit: may want to name this something like `ensure_ended_and_exported` to signal that it can be called multiple times, else it may look like a bug that it's called on `end` and `Drop`